File: frontend/src/api/chats.ts
const API = import.meta.env.VITE_API_URL || "";
import { apiFetch } from "./http";

export interface ChatItem {
  id: string;
  telegram_chat_id: number;
  title: string | null;
  is_monitored: boolean;
  chat_type: string | null;
  username: string | null;
  parent_telegram_chat_id: number | null;
  has_photo: boolean;
  created_at: string;
  updated_at: string | null;
}

export interface ChatsStatus {
  monitored_count: number;
  total_count: number;
  has_monitored: boolean;
}

export async function fetchChatsStatus(): Promise<ChatsStatus> {
  const r = await apiFetch(`${API}/api/chats/status`);
  if (!r.ok) throw new Error("Failed to load chat status");
  return r.json();
}

export async function fetchChats(monitoredOnly = false): Promise<{
  items: ChatItem[];
  total: number;
  monitored_count: number;
}> {
  const q = monitoredOnly ? "?monitored_only=true" : "";
  const r = await apiFetch(`${API}/api/chats${q}`);
  if (!r.ok) throw new Error("Failed to load chats");
  return r.json();
}

export async function syncTelegramChats(): Promise<{ synced: number; items: ChatItem[] }> {
  const r = await apiFetch(`${API}/api/chats/sync`, { method: "POST" });
  if (!r.ok) {
    const err = await r.json().catch(() => ({}));
    throw new Error((err as { detail?: string }).detail || "Sync failed");
  }
  return r.json();
}

export async function saveChatSelection(telegramChatIds: number[]): Promise<void> {
  const r = await apiFetch(`${API}/api/chats/selection`, {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ telegram_chat_ids: telegramChatIds }),
  });
  if (!r.ok) {
    const err = await r.json().catch(() => ({}));
    throw new Error((err as { detail?: string }).detail || "Save failed");
  }
}

export function chatAvatarUrl(chatId: string): string {
  return `${API}/api/chats/${chatId}/avatar`;
}

export const CHAT_TYPE_LABELS: Record<string, string> = {
  private: "Личный",
  group: "Группа",
  supergroup: "Супергруппа",
  channel: "Канал",
  unknown: "Чат",
};

File: backend/app/schemas/chats.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ChatOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    telegram_chat_id: int
    title: str | None
    is_monitored: bool
    chat_type: str | None
    username: str | None
    parent_telegram_chat_id: int | None
    has_photo: bool = False
    created_at: datetime
    updated_at: datetime | None = None


class ChatList(BaseModel):
    items: list[ChatOut]
    total: int
    monitored_count: int


class ChatStatusOut(BaseModel):
    monitored_count: int
    total_count: int
    has_monitored: bool


class ChatSyncOut(BaseModel):
    synced: int
    items: list[ChatOut]


class ChatSelectionIn(BaseModel):
    telegram_chat_ids: list[int] = Field(default_factory=list)
